src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -152,23 +161,89 @@ await _transport.SendMessageAsync(new JsonRpcError
 
     private async Task HandleMessageAsync(IJsonRpcMessage message, CancellationToken cancellationToken)
     {
-        switch (message)
+        using Activity? activity = s_activitySource.StartActivity("HandlingMessage");
```

Comment:
Is there a better name than `HandlingMessage` that will provide more context when somebody is looking at a trace. Presumably this is for the server so having a name that corresponds to the call would be useful?

Reply:
I put the information about the request, what method is being used, what tool is being called, etc., into tags on the activity. Is changing the name of the activity preferable? e.g. do we name activities elsewhere in the system based on url / query string / etc.?

Reply:
The name is less important than the description that you can also set. That is what shows in the trace viewer in the Aspire  Dashboard.